testsuite/tests/mgc/dnspolicy/health_check/conftest.py

```diff
@@ -0,0 +1,47 @@
+"""Conftest for DNSPolicy health checks"""
+import time
+import pytest
+
+from testsuite.openshift.objects.gateway_api.gateway import MGCGateway
+from testsuite.openshift.objects.dnspolicy import DNSPolicy
+
+
+@pytest.fixture(scope="module")
+def upstream_gateway(request, openshift, blame, module_label, initial_host):
+    """
+    Creates and returns configured and ready upstream Gateway with FQDN hostname
+    Health checks available only with Fully Qualified Domain Names in gateway (no wildcards are allowed)
+    """
+    upstream_gateway = MGCGateway.create_instance(
+        openshift=openshift,
+        name=blame("mgc-gateway"),
+        gateway_class="kuadrant-multi-cluster-gateway-instance-per-cluster",
+        hostname=initial_host,
+        tls=False,
+        placement="http-gateway",
+        labels={"app": module_label},
+    )
+    request.addfinalizer(upstream_gateway.delete)
+    upstream_gateway.commit()
+
+    return upstream_gateway
+
+
+@pytest.fixture(scope="module")
+def dns_policy(blame, upstream_gateway, module_label, health_check):
+    """DNSPolicy with health check"""
+    policy = DNSPolicy.create_instance(
+        upstream_gateway.openshift,
+        blame("dns"),
+        upstream_gateway,
+        healthCheck=health_check,
+        labels={"app": module_label},
+    )
+    return policy
+
+
+@pytest.fixture(scope="module")
+def dns_health_probe(dns_policy, route):  # pylint: disable=unused-argument
+    """Wait for health check to start monitoring endpoint and return according DNSHealthCheckProbe object"""
+    time.sleep(10)
```

Comment:
why the sleep?

Reply:
There is some time until the DNSHealthCheckProbe object will be created. It is a possibility to try getting this object and backoff on failure multiple times, or refactor it later - your call

Reply:
Is there currently a way to know the object was created? e.g. Some reference in the annotation or label?

Reply:
Nothing that I'm aware of

Reply:
I believe this existing issue reflects your question: https://github.com/Kuadrant/multicluster-gateway-controller/issues/641